src/Services/AllergyIntoleranceService.php

```diff
@@ -88,7 +88,7 @@ public function search($search, $isAndCondition = true)
             FROM users
             -- US CORE only allows physicians or patients to be our allergy recorder
             -- so we will filter out anyone who is not actually a practitioner (May 14th 2021)
-            WHERE users.npi IS NOT NULL -- we only want actual physicians here rather than all users
+            WHERE users.authorized = 1
```

Comment:
@adunsulag and @stephenwaite ,
Guessing this will break inferno since i recall weirdness where allergies needed to be associated with actual physicians.

The problem here is that allergies many time are entered in by staff and we really need to ensure that all allergies are shown/reported (ie. high risk of hurting patients if do not).

Would there be a way to get around this?

Reply:
doesn't seem practical or realistic for inferno to require only physicians to enter allergies

registered nurses are a good example of a clinician who wouldn't have an npi

Reply:
registered nurses probably are under the purview of a physician or other practitioner who does have an npi so that is a fallback

that might mean reworking the allergy issue input as well then

Reply:
@stephenwaite looking at the specification, it is nullable.  So we can remove the NPI check but what we will need to do is return the NPI number as part of the service and then check in the FHIR service for the NPI number.  If the NPI number is null we don't populate the recorder.  We'd want to test it all with inferno, but I think that will address this issue.


Essentially in FHIRAllergyIntoleranceService.php:L166-L170 it becomes:
```php
if (isset($dataRecord['practitioner']) && !empty($dataRecord['practitioner_npi']) {
            $recorder = new FHIRReference();
            $recorder->setReference('Practitioner/' . $dataRecord['practitioner']);
            $allergyIntoleranceResource->setRecorder($recorder);
        }
```

AllergyIntoleranceService.php then becomes:
```php
$sql = "SELECT lists.*,
        lists.pid AS patient_id,
        lists.title,
        lists.comments,
        practitioners.uuid as practitioner,
        practitioners.practitioner_npi,
        practitioners.practitioner_uuid,
        // .....
        LEFT JOIN (
            select
            users.uuid
            ,users.uuid AS practitioner_uuid
            ,users.npi AS practitioner_npi,
            ,users.username
            ,users.facility AS organization
            FROM users
            -- US CORE only allows physicians or patients to be our allergy recorder
            -- so we will filter out anyone who is not actually a practitioner (May 14th 2021)
            WHERE users.npi IS NOT NULL -- we only want actual physicians here rather than all users
        ) practitioners ON practitioners.username = lists.user
        // .....
```

Reply:
what's the link to the spec?

Reply:
https://hl7.org/fhir/us/core/STU3.1.1/StructureDefinition-us-core-allergyintolerance.html  They have the differential table, but if you click on the full view it will show the recorder property.

Reply:
thanks @adunsulag